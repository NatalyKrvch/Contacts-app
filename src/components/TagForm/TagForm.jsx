import { useState } from 'react'
import { Button } from 'react-bootstrap'
import TextInput from 'components/TextInput/TextInput'

const TagForm = ({ onAddTags }) => {
  const [tagInput, setTagInput] = useState('')

  const handleAddTags = () => {
    const newTags = tagInput
      .split(',')
      .map((tag) => tag.trim())
      .filter((tag) => tag)

    if (newTags.length > 0) {
      onAddTags(newTags)
      setTagInput('')
    }
  }

  return (
    <div>
      <TextInput
        value={tagInput}
        onChange={setTagInput}
        placeholder="Add new Tags separated by commas"
      />
      <Button onClick={handleAddTags} className="mt-2 w-100">
        Add Tags
      </Button>
    </div>
  )
}

export default TagForm
