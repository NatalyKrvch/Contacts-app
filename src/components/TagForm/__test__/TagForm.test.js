import { render, screen, fireEvent } from '@testing-library/react';
import TagForm from '../TagForm';

describe('TagForm', () => {
  const mockOnAddTags = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders TagForm correctly', () => {
    const { container } = render(<TagForm onAddTags={mockOnAddTags} />);
    
    expect(screen.getByPlaceholderText('Add new Tags separated by commas')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /add tags/i })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('updates input value when typing', () => {
    render(<TagForm onAddTags={mockOnAddTags} />);
    
    const input = screen.getByPlaceholderText('Add new Tags separated by commas');
    fireEvent.change(input, { target: { value: 'tag1, tag2' } });
    
    expect(input.value).toBe('tag1, tag2');
    expect(input).toMatchSnapshot();
  });

  it('does not call onAddTags if input is empty', () => {
    render(<TagForm onAddTags={mockOnAddTags} />);
    
    fireEvent.click(screen.getByRole('button', { name: /add tags/i }));
    
    expect(mockOnAddTags).not.toHaveBeenCalled();
    expect(screen.getByPlaceholderText('Add new Tags separated by commas')).toMatchSnapshot();
  });
});
