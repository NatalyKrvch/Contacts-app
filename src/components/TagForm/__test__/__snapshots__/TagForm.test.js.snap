// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagForm does not call onAddTags if input is empty 1`] = `
<input
  class="form-control"
  placeholder="Add new Tags separated by commas"
  type="text"
  value=""
/>
`;

exports[`TagForm renders TagForm correctly 1`] = `
<div>
  <div>
    <div>
      <input
        class="form-control"
        placeholder="Add new Tags separated by commas"
        type="text"
        value=""
      />
      <div
        class="invalid-feedback"
      />
    </div>
    <button
      class="mt-3 w-100 btn btn-success"
      type="button"
    >
      Add Tags
    </button>
  </div>
</div>
`;

exports[`TagForm updates input value when typing 1`] = `
<input
  class="form-control"
  placeholder="Add new Tags separated by commas"
  type="text"
  value="tag1, tag2"
/>
`;
